library-udf/src/test/java/org/apache/iotdb/library/anomaly/AnomalyTests.java

```diff
@@ -0,0 +1,512 @@
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      resultSet.last();
+      int resultSetLength = resultSet.getRow();
+      Assert.assertEquals(resultSetLength, 1);
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @Test
+  public void testRange3() {
+    String sqlStr =
+        "select range(d2.s1,\"lower_bound\"=\"-5.0\",\"upper_bound\"=\"5.0\") from root.vehicle";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      resultSet.last();
+      int resultSetLength = resultSet.getRow();
+      Assert.assertEquals(resultSetLength, 1);
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @Test
+  public void testRange4() {
+    String sqlStr =
+        "select range(d2.s2,\"lower_bound\"=\"-5.0\",\"upper_bound\"=\"5.0\") from root.vehicle";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      resultSet.last();
+      int resultSetLength = resultSet.getRow();
+      Assert.assertEquals(resultSetLength, 1);
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @Test
+  public void testTwoSidedFileter1() {
+    String sqlStr = "select TwoSidedFilter(d1.s1, 'len'='3', 'threshold'='0.3') from root.vehicle";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      resultSet.last();
+      int resultSetLength = resultSet.getRow();
+      assert resultSetLength <= 10;
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @Test
+  public void testTwoSidedFileter2() {
+    String sqlStr = "select TwoSidedFilter(d1.s2, 'len'='3', 'threshold'='0.3') from root.vehicle";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      resultSet.last();
+      int resultSetLength = resultSet.getRow();
+      assert resultSetLength <= 10;
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @Test
+  public void testTwoSidedFileter3() {
+    String sqlStr = "select TwoSidedFilter(d2.s1, 'len'='3', 'threshold'='0.3') from root.vehicle";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet = statement.executeQuery(sqlStr);
+      resultSet.last();
+      int resultSetLength = resultSet.getRow();
+      assert resultSetLength <= 10;
```

Comment:
As we have discussed previously, please check the equalivent between expected and acutal data returned, not only row counts.
And do not use `assert` keyword.